internal/storage/main.go

```diff
@@ -0,0 +1,102 @@
+package storage
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/url"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+)
+
+func (s *Storage) GetImageBase64(object *url.URL) (*string, error) {
+	spacesURL, err := parseDOSpacesURL(object)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url [%s]: %w", object.String(), err)
+	}
+
+	output, err := s.client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(spacesURL.Bucket),
+		Key:    aws.String(spacesURL.Key),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get object meta: %w", err)
+	}
+	defer output.Body.Close()
+
+	imageBytes, err := io.ReadAll(output.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read image: %w", err)
+	}
+
+	imageBase64 := base64.StdEncoding.EncodeToString(imageBytes)
+
+	return &imageBase64, nil
+}
+
+func (s *Storage) ValidateImage(object *url.URL) error {
+	spacesURL, err := parseDOSpacesURL(object)
+	if err != nil {
+		return fmt.Errorf("failed to parse url [%s]: %w", object.String(), err)
+	}
+
+	if func() error {
```

Comment:
holy crap, I know Go allows this, but maybe we write it normally? just "found:=false; for { ... }; if !found return Err... Or at least with `func() error` outside `if`